lib/ejs.js

```diff
@@ -330,43 +357,39 @@ exports.render = function (template, d, o) {
  */
 
 exports.renderFile = function () {
-  var args = Array.prototype.slice.call(arguments);
-  var filename = args.shift();
-  var cb = args.pop();
-  var data = args.shift() || {};
-  var opts = args.pop() || {};
-  var optsKeys =_OPTS.slice();
-  var result;
-
-  // Don't pollute passed in opts obj with new vals
-  opts = utils.shallowCopy({}, opts);
-
-  // We don't allow 'cache' option to be passed in the data obj
-  // for the normal `render` call, but this is where Expres puts it
-  // so we make an exception for `renderFile`
-  optsKeys.push('cache');
-
-  // No options object -- if there are optiony names
-  // in the data, copy them to options
-  if (arguments.length == 3) {
-    // Express 4
-    if (data.settings && data.settings['view options']) {
-      utils.shallowCopyFromList(opts, data.settings['view options'], optsKeys);
+  var argsLen = arguments.length;
```

Comment:
I'd personally just rather use `arguments.length` everywhere this is needed. JS object lookups are so fast, is optimization really worth it? Just my 2 cents.